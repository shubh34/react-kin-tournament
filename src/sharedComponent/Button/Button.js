import './Button.scss';
import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ name, onClick, className }) => (
	<button type="button" className={`c-button ${className}`} onClick={onClick}>
		{name}
	</button>
);


Button.propTypes = {
	name: PropTypes.string.isRequired,
	className: PropTypes.string,
	onClick: PropTypes.func.isRequired,
};

export default Button;
