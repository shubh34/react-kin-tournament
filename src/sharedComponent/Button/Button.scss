.c-button {
    padding: 24px;
    font-size: 32px;
    font-family: sans-serif;
    cursor: pointer;
    outline:none;
    border:none;
    &--active {
        font-weight:bold;
    }
}