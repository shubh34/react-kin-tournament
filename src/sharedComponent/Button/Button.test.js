import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
	it('renders without crashing', () => {
		const wrapper = shallow(<Button name="button" onClick={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should call onClick on button click', () => {
		const onClick = jest.fn();
		const wrapper = shallow(<Button name="button" onClick={onClick} />);
		wrapper.find('button').simulate('click');
		expect(onClick).toBeCalled();
	});
});
