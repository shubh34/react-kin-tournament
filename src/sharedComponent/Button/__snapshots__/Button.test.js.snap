// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<button
  className="c-button undefined"
  id="button"
  name="button"
  onClick={[MockFunction]}
  type="button"
>
  button
</button>
`;
